tests/TestCase/AuthorizationServiceTest.php

```diff
@@ -0,0 +1,135 @@
+<?php
+/**
+ * CakePHP(tm) : Rapid Development Framework (https://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ * @link          https://cakephp.org CakePHP(tm) Project
+ * @since         1.0.0
+ * @license       https://opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Authorization\Test\TestCase;
+
+use Authorization\AuthorizationService;
+use Authorization\IdentityDecorator;
+use Authorization\Policy\BeforePolicyInterface;
+use Authorization\Policy\Exception\MissingMethodException;
+use Authorization\Policy\ResolverInterface;
+use Cake\TestSuite\TestCase;
+use TestApp\Authorization\Model\Entity\Article;
+use TestApp\Authorization\Policy\Model\Entity\Article as ArticlePolicy;
+
+class AuthorizationServiceTest extends TestCase
+{
+    public function testCan()
+    {
+        $resolver = $this->createMock(ResolverInterface::class);
+        $entity = new Article();
+        $policy = new ArticlePolicy();
+
+        $resolver->expects($this->once())
+            ->method('getPolicy')
+            ->with($entity)
+            ->willReturn($policy);
```

Comment:
Instead of a mock you could use the `MapResolver`. It would save LOC and give us integration tests for 'free'.

Reply:
I have a habit of testing thing in isolation as much as possibke, but it is a good point. I'll change this.